api/server/router/swarm/cluster_routes.go

```diff
@@ -63,15 +64,17 @@ func (sr *swarmRouter) updateCluster(ctx context.Context, w http.ResponseWriter,
 	rawVersion := r.URL.Query().Get("version")
 	version, err := strconv.ParseUint(rawVersion, 10, 64)
 	if err != nil {
-		return fmt.Errorf("Invalid swarm version '%s': %s", rawVersion, err.Error())
+		err := fmt.Errorf("Invalid swarm version '%s': %v", rawVersion, err)
+		return errors.NewBadRequestError(err)
```

Comment:
This errors package is pretty awful. I am not sure that I would like see it's proliferation. Could we at least follow the conventions of https://godoc.org/github.com/pkg/errors and have these implement `Cause`?

I also don't get why we have a constructor for each one: should be `NewAPIError(http.StatusBadRequest, err)`:

```go
func NewAPIError(status int, cause error) error
```

The returned error would implement `causer` and `StatusCode`.

I know fixing this package is outside of the scope of this change, but I don't want to see the usage of such a poorly designed package increase.